x-pack/solutions/security/plugins/security_solution/public/common/hooks/timeline/use_timeline_api.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import type { Filter } from '@kbn/es-query';
+import { useCreateTimeline } from '../../../timelines/hooks/use_create_timeline';
+import { updateProviders, setFilters } from '../../../timelines/store/actions';
+import { SourcererScopeName } from '../../../sourcerer/store/model';
+import type { DataProvider } from '../../../../common/types';
+import { sourcererSelectors } from '../../store';
+import { sourcererActions } from '../../store/actions';
+import { inputsActions } from '../../store/inputs';
+import { InputsModelId } from '../../store/inputs/constants';
+import type { TimeRange } from '../../store/inputs/model';
+import { TimelineId } from '../../../../common/types/timeline';
+import { TimelineTypeEnum } from '../../../../common/api/timeline';
+
+interface OpenTimelineArgs {
```

Comment:
if you're ok renaming the hook to `useInvestigateInTimeline` we can rename this to `InvestigateInTimelineArgs`